### ISSUE TITLE
reduce_sum和softmax有bug?

### ISSUE THREAD

**Issue by zhangyafeikimi** [hexiangnan/attentional_factorization_machine#19]
self.attention_relu = tf.reduce_sum(tf.multiply(self.weights['attention_p'], tf.nn.relu(self.attention_mul + \
                    self.weights['attention_b'])), 2, keep_dims=True) # None * (M'*(M'-1)) * 1
                self.attention_out = tf.nn.softmax(self.attention_relu)

我认为keep_dims应该为False.

keep_dims如果为True, attention_relu的shape是(batch, m*(m-1)/2, 1)
接下来的softmax会在attention_relu的最后一个axis上做归一化, 这样是错误的.

**Comment by zhangyafeikimi** [comment]
单纯改成False也不行, 后面的计算也要相应的修改

**Comment by zzx2017** [comment]
tf.nn.softmax中reduce_sum的默认维度是dim=-1，改成tf.nn.softmax(self.attention_relu，dim=1)。

**Comment by zhangyafeikimi** [comment]
this repo is dead?
